parlai/tasks/empathetic_dialogues/agents.py

```diff
@@ -139,5 +139,65 @@ def share(self):
         return shared
 
 
+class EmotionClassificationTeacher(EmpatheticDialogueTeacher):
+    """
+    Class for detecting the emotion based on the utterance.
+    """
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        parser = parser.add_argument_group('Emotion Classification Args')
+        parser.add_argument(
+            '--single-turn',
+            type='bool',
+            default=True,
+            help='Single turn classification task',
+        )
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        self.single_turn = opt['single_turn']
+        if not shared and self.single_turn:
+            self._make_single_turn()
+
+    def num_episodes(self):
+        return len(self.data)
+
+    def num_examples(self):
+        if not self.single_turn:
+            return super().num_examples()
+        return len(self.data)
+
+    def _make_single_turn(self):
+        new_data = []
+        for ep in self.data:
+            for ex in ep:
+                new_data.append(ex)
+        self.data = new_data
+
+    def get(self, episode_idx, entry_idx=0):
+        if not self.single_turn:
+            ep = self.data[episode_idx]
+            i = entry_idx * 2
+            ex = ep[i]
+            episode_done = i >= (len(ep) - 2)
+        else:
+            ex = self.data[episode_idx]
+            episode_done = True
+
+        return {
+            'situation': ex[3],
```

Comment:
O_o

Reply:
lol i was just copying the original set-up, but i can put them in order if you'd prefer